wrapanapi/containers/providers/rhopenshift.py

```diff
@@ -0,0 +1,874 @@
+        else:
+            raise ValueError("This name is not found among deployment configs or stateful sets")
+        self.logger.info("scaling entity %s to %s replicas", name, replicas)
+        wait_for(check_scale_value, num_sec=wait, fail_condition=lambda val: val != replicas)
+
+    def get_project_by_name(self, project_name):
+        """Returns only the selected Project object"""
+        return Project(self, project_name)
+
+    def is_vm_running(self, vm_name):
+        """Emulates check is vm(appliance) up and running
+
+        Args:
+            vm_name: project(namespace) name
+        Return: True/False
+        """
+        logger.info("checking all pod statuses")
+        for pod_name in self.required_project_pods:
+            if self.is_deployment_config(name=pod_name, namespace=vm_name):
+                dc = self.o_api.read_namespaced_deployment_config(name=pod_name, namespace=vm_name)
+                status = dc.status.ready_replicas
+            elif self.is_stateful_set(name=pod_name, namespace=vm_name):
+                pods = self.k_api.list_namespaced_pod(namespace=vm_name,
+                                                      label_selector='name={n}'.format(n=pod_name))
+                pod_stats = [pod.status.container_statuses[-1].ready for pod in pods.items]
+                status = all(pod_stats)
+            else:
+                raise ValueError("No such pod name among StatefulSets or Stateless Pods")
+
+            if status and int(status) > 0:
+                logger.debug("pod {name} looks up and running".format(name=pod_name))
+                continue
+            else:
+                logger.debug("pod {name} isn't up yet".format(name=pod_name))
+                return False
+        # todo: check url is available + db is accessable
+        return True
+
+    def list_deployment_config_names(self, namespace):
+        """Extracts and returns list of Deployment Config names
+
+        Args:
+            namespace: project(namespace) name
+        Return: (list) deployment config names
+        """
+        dcs = self.o_api.list_namespaced_deployment_config(namespace=namespace)
+        return [dc.metadata.name for dc in dcs.items]
+
+    def list_stateful_set_names(self, namespace):
+        """Returns list of Stateful Set names
+
+        Args:
+            namespace: project(namespace) name
+        Return: (list) stateful set names
+        """
+        st_api = self.kclient.AppsV1beta1Api()
+        sts = st_api.list_namespaced_stateful_set(namespace=namespace)
+        return [st.metadata.name for st in sts.items]
+
+    def is_deployment_config(self, namespace, name):
+        """Checks whether passed name belongs to deployment configs in appropriate namespace
+
+        Args:
+            namespace: project(namespace) name
+            name: entity name
+        Return: True/False
+        """
+        return True if name in self.list_deployment_config_names(namespace=namespace) else False
```

Comment:
**REQUIRED**: You can drop the ternary here and just return the condition:

```python
return name in self.list_deployment_config_names(namespace=namespace)
```

The same in `is_stateful_set`

Reply:
agreed, my bad